ui/app/src/utils/path.ts

```diff
@@ -205,3 +205,26 @@ export function getCodeEditorSrc({
   });
   return `${authoringBase}/legacy/form${qs}`;
 }
+
+export function expandPathMacros(path: string) {
+  // TODO: Omitted values 'objectId', 'objectGroupId', 'parentPath'
+
+  const currentDate = new Date();
+  if (path.includes('{year}') || path.includes('{yyyy}')) {
+    let year = String(currentDate.getFullYear());
+    path = path.replace('{year}', year);
+    path = path.replace('{yyyy}', year);
+  }
```

Comment:
```suggestion
path = path.replace(/{(year|yyyy)}/g, String(currentDate.getFullYear()));
```
same deal for the ones below. No need for if.